### ISSUE TITLE
pyfastx.Fasta leaking memory?

### ISSUE THREAD

**Issue by hmwill** [lmdu/pyfastx#6]
I am opening a  [large fasta file](ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/uniparc/uniparc_active.fasta.gz) via

```
import pyfastx
fasta_sequences = pyfastx.Fasta('data/uniparc_active.fasta', build_index=True)
```
and I am seeing memory utilization in top increase until 100%. 

Is this intended behavior, or is there a memory leak in the library?

Thanks,
Martin

**Comment by hmwill** [comment]
Looking at the code in index.c. Should the last parameter rather be `free` instead of `NULL` in order to release the allocated buffer for the string?

```
				sqlite3_bind_text(stmt, 2, chrom, -1, NULL);
				...
				sqlite3_bind_text(stmt, 9, description, -1, NULL);
```

The sqlite documentation only seems to talk of `SQLITE_STATIC`, but its definition is actually

```
#define SQLITE_STATIC      ((sqlite3_destructor_type)0)
```

so passing `NULL` means don't call the destructor.

Also, if I understand the code correctly, the whole file is processed as a single transaction.

**Comment by lmdu** [comment]
Thank you for your reporting this issue. 

I have downloaded the uniparc_active.fasta.gz to test pyfastx. This file contains **305,146,894** sequences.  Using the current version of pyfastx to build index, it has  consumed **3345.46 seconds** and **31.31Gb** memory. According to your suggestion, I replaced the last parameter of sqlite3_bind_text to `free` and redo the test, it has consumed **3019.91 seconds** and  **2.7Gb** memory in a single transaction. This is a amazing improvement for parsing huge fasta file. If the file was decompressed prior to building index  , the consumed time will be reduced. 

I will added this improvement in later version.
Thanks again!
